src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -14,35 +14,81 @@ namespace Microsoft.Health.Events.Telemetry.Exceptions
 {
     public static class EventHubExceptionTelemetryProcessor
     {
+        private static readonly EventHubConfigurationExceptionTelemetryProcessor _exceptionTelemetryProcessor = new EventHubConfigurationExceptionTelemetryProcessor();
+
         public static void ProcessException(
             Exception exception,
             ITelemetryLogger logger,
-            bool shouldLogMetric = true,
             string errorMetricName = null)
         {
             EnsureArg.IsNotNull(exception, nameof(exception));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            logger.LogError(exception);
+            var ex = CustomizeException(exception);
+
+            logger.LogError(ex);
 
-            if (shouldLogMetric)
+            if (ex.Equals(exception))
             {
-                logger.LogMetric(EventMetrics.HandledException(errorMetricName ?? GetErrorMetricName(exception), ConnectorOperation.Setup), 1);
+                logger.LogMetric(
+                    EventMetrics.HandledException(
+                        errorMetricName ?? $"{ErrorType.EventHubError}{EventHubErrorCode.GeneralError}",
+                        ConnectorOperation.Setup),
+                    1);
+            }
+            else
+            {
+                _exceptionTelemetryProcessor.HandleException(ex, logger, ConnectorOperation.Setup);
             }
         }
 
-        private static string GetErrorMetricName(Exception exception)
+        public static Exception CustomizeException(Exception exception)
         {
+            string message = exception.Message;
+
             switch (exception)
             {
                 case EventHubsException _:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.OperationError}";
+                    EventHubsException eventHubsException = (EventHubsException)exception;
+                    switch (eventHubsException.Reason)
+                    {
+                        case EventHubsException.FailureReason.ConsumerDisconnected:
+                            message = "Verify that the provided Event Hub's consumer group is not already receiving data for another IoT connector or Azure resource.";
```

Comment:
Maybe something we can consider as a backlog item, but ideally these strings should be localized.

Reply:
Would you be able to create this tracking backlog item?